src/main/java/mate/academy/bookstore/controller/BookController.java

```diff
@@ -0,0 +1,50 @@
+package mate.academy.bookstore.controller;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.BookDto;
+import mate.academy.bookstore.dto.CreateBookRequestDto;
+import mate.academy.bookstore.service.BookService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
+public class BookController {
+    private final BookService bookService;
+
+    @PostMapping
+    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+        return bookService.save(bookDto);
+    }
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.findById(id);
+    }
+
+    @GetMapping("/by-title")
+    public List<BookDto> findByTitle(@RequestParam String title) {
+        return bookService.findByTitle(title);
+    }
```

Comment:
Remove this endpoint and all relevant code. You will implement the search endpoint in the next task